boa/src/builtins/value/mod.rs

```diff
@@ -149,6 +151,103 @@ impl Value {
 
         Self::object(object)
     }
+
+    /// Convert from a JSON value to a JS value
+    pub fn from_json(json: JSONValue, interpreter: &mut Interpreter) -> Self {
+        match json {
+            JSONValue::Number(v) => {
+                if let Some(Ok(integer_32)) = v.as_i64().map(i32::try_from) {
+                    Self::integer(integer_32)
+                } else {
+                    Self::rational(v.as_f64().expect("Could not convert value to f64"))
+                }
+            }
+            JSONValue::String(v) => Self::string(v),
+            JSONValue::Bool(v) => Self::boolean(v),
+            JSONValue::Array(vs) => {
+                let mut new_obj = Object::default();
+                let length = vs.len();
+                for (idx, json) in vs.into_iter().enumerate() {
+                    new_obj.properties.insert(
+                        idx.to_string(),
+                        Property::default()
+                            .value(Self::from_json(json, interpreter))
+                            .writable(true)
+                            .configurable(true),
+                    );
+                }
+                new_obj.properties.insert(
+                    "length".to_string(),
+                    Property::default().value(Self::from(length)),
+                );
+                Self::object(new_obj)
+            }
+            JSONValue::Object(obj) => {
+                let new_obj = Value::new_object(Some(&interpreter.realm.global_obj));
+                for (key, json) in obj.into_iter() {
+                    let value = Self::from_json(json, interpreter);
+                    new_obj.set_property(
+                        key,
+                        Property::default()
+                            .value(value)
+                            .writable(true)
+                            .configurable(true),
+                    );
+                }
+                new_obj
+            }
+            JSONValue::Null => Self::null(),
+        }
+    }
+
+    /// Conversts the `Value` to `JSON`.
+    pub fn to_json(&self, interpreter: &mut Interpreter) -> Result<JSONValue, Value> {
+        match *self.data() {
+            ValueData::Null => Ok(JSONValue::Null),
+            ValueData::Boolean(b) => Ok(JSONValue::Bool(b)),
+            ValueData::Object(ref obj) => {
+                if obj.borrow().kind == ObjectKind::Array {
+                    let mut arr: Vec<JSONValue> = Vec::new();
+                    obj.borrow().properties.keys().for_each(|k| {
+                        if k != "length" {
+                            let value = self.get_field(k.to_string());
+                            if value.is_undefined() || value.is_function() {
+                                arr.push(JSONValue::Null);
+                            } else {
+                                arr.push(
+                                    self.get_field(k.to_string()).to_json(interpreter).unwrap(),
```

Comment:
Here too, we should use `?` instead of `unwrap()`.

Reply:
I'm struggling with this one. It doesn't seem that we can use the `?` operator because we're inside of a for_each.

Reply:
I would just use a `for`, it seems more accurate in this case:
```rust
for k in obj.borrow().properties.keys() {
    // ...
}
```

Reply:
:heavy_check_mark: